src/Vision/IdentityDocumentParser.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Vision;
+
+use App\Entity\Adherent;
+use Cocur\Slugify\Slugify;
+
+class IdentityDocumentParser
+{
+    private $slugify;
+
+    public function __construct(Slugify $slugify)
+    {
+        $this->slugify = $slugify;
+    }
+
+    public function match(Adherent $adherent, ImageAnnotations $imageAnnotations): bool
+    {
+        return $this->hasFirstName($imageAnnotations, $adherent->getFirstName())
+            && $this->hasLastName($imageAnnotations, $adherent->getLastName())
+            && $this->hasDateOfBirth($imageAnnotations, $adherent->getBirthdate())
+        ;
+    }
+
+    public function hasFirstName(ImageAnnotations $imageAnnotations, string $firstName): bool
+    {
+        if ($imageAnnotations->isFrenchNationalIdentityCard()) {
+            preg_match('/Prénom( )?\(s\)( )?:( )?(?<first_names>.+)\\n/', $imageAnnotations->getText(), $matches);
```

Comment:
I had updated this regex in my previous PR to fix an issue with parentheses. Don't you want to reuse it ?